### ISSUE TITLE
Missing type attribute for <button> elements

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8213]
### Issue Summary

This is a report from experimental automated tests which I have only partly confirmed.

There are a number of `<button>` elements in our templates which are missing a `type` attribute to indicate the button’s type – either `button` or `submit`. This attribute is optional, but defaults to `submit`, which can be a source of bugs. As such, we always prefer to systematically add an explicit `type` attribute.

### Steps to Reproduce

Here are the flagged templates:

```
wagtail/admin/templates/wagtailadmin/account/settings_panels/avatar.html
wagtail/admin/templates/wagtailadmin/bulk_actions/footer.html
wagtail/contrib/forms/templates/wagtailforms/index_submissions.html
```

**Comment by 0saurabh0** [comment]
finally some good first issues.
hey @thibaudcolas i want to work on this, any points for getting started.

**Comment by thibaudcolas** [comment]
@0saurabh0 sorry, this is already being worked on by @jessemenn 😕.

**Comment by 0saurabh0** [comment]
guess, i have to pick another one then ??
